src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -140,8 +142,13 @@ func NewStatefulSet(
 		adminAPIClientFactory,
 		decommissionWaitInterval,
 		logger.WithValues("Kind", statefulSetKind()),
+		adminAPITimeout,
```

Comment:
so there's two sets of the timeouts - admin api and metrics and if the metrics is 0 we use admin api timeout? should we document that somehow?

Reply:
also where is this timeout enforced? Don't see it 🤔

Reply:
My bad. I forgot to change that.

Reply:
>  admin api and metrics and if the metrics is 0 we use admin api timeout? should we document that somehow?

I will add the description for the `metrics-timeout` program flag.